src/kv/test/kv_dynamic_tables.cpp

```diff
@@ -0,0 +1,292 @@
+    REQUIRE(v.has_value());
+    view3->put("foo", "baz");
+    REQUIRE(view3->get("foo").value() == "baz");
+
+    REQUIRE(tx3.commit() == kv::CommitSuccess::OK);
+  }
+
+  {
+    REQUIRE(kv_store.get<MapTypes::StringString>(map_name) != nullptr);
+  }
+
+  {
+    INFO("Subsequent transactions over dynamic map are persisted");
+    auto tx4 = kv_store.create_tx();
+    auto view4 = tx4.get_view2<MapTypes::StringString>(map_name);
+    const auto v = view4->get("foo");
+    REQUIRE(v.has_value());
+    REQUIRE(v.value() == "baz");
+  }
+}
+
+TEST_CASE(
+  "Dynamic table visibility by version" * doctest::test_suite("dynamic"))
+{
+  kv::Store kv_store;
+
+  auto encryptor = std::make_shared<kv::NullTxEncryptor>();
+  kv_store.set_encryptor(encryptor);
+
+  auto& static_map = kv_store.create<MapTypes::StringString>("static_map");
+
+  constexpr auto map_name = "dynamic_map";
+
+  auto tx1 = kv_store.create_tx();
+  auto tx2 = kv_store.create_tx();
+  auto tx3 = kv_store.create_tx();
+  auto tx4 = kv_store.create_tx();
+
+  auto view1 = tx1.get_view2<MapTypes::StringString>(map_name);
+  view1->put("foo", "bar");
+
+  // Map created in tx1 is not visible
+  auto view2 = tx2.get_view2<MapTypes::StringString>(map_name);
+  REQUIRE(!view2->get("foo").has_value());
+
+  // tx3 takes a read dependency at an early version, before the map is visible
+  auto view3_static = tx3.get_view(static_map);
+
+  REQUIRE(tx1.commit() == kv::CommitSuccess::OK);
+
+  // Even after commit, the new map is not visible to tx3 because it is reading
+  // from an earlier version
+  auto view3 = tx3.get_view2<MapTypes::StringString>(map_name);
+  REQUIRE(!view3->get("foo").has_value());
+
+  // Map created in tx1 is visible, because tx4 first _reads_ (creates a
+  // view) after tx1 has committed
+  auto view4 = tx4.get_view2<MapTypes::StringString>(map_name);
+  REQUIRE(view4->get("foo").has_value());
+}
+
+TEST_CASE("Mixed map dependencies" * doctest::test_suite("dynamic"))
+{
+  kv::Store kv_store;
+
+  auto encryptor = std::make_shared<kv::NullTxEncryptor>();
+  kv_store.set_encryptor(encryptor);
+
+  constexpr auto dynamic_map_a = "dynamic_map_a";
+  constexpr auto dynamic_map_b = "dynamic_map_b";
+
+  auto& static_map = kv_store.create<MapTypes::StringString>("static_map");
+
+  SUBCASE("Parallel independent map creation")
+  {
+    auto tx1 = kv_store.create_tx();
+    auto tx2 = kv_store.create_tx();
+
+    auto view1 = tx1.get_view2<MapTypes::NumString>(dynamic_map_a);
+    auto view2 = tx2.get_view2<MapTypes::StringNum>(dynamic_map_b);
+
+    view1->put(42, "hello");
+    view2->put("hello", 42);
+
+    REQUIRE(tx1.commit() == kv::CommitSuccess::OK);
+    REQUIRE(tx2.commit() == kv::CommitSuccess::OK);
+  }
+
+  SUBCASE("Map creation blocked by standard conflict")
+  {
+    constexpr auto key = "foo";
+    auto tx1 = kv_store.create_tx();
+    {
+      auto view1 = tx1.get_view(static_map);
+      const auto v = view1->get(key);
+      if (!v.has_value())
```

Comment:
Why this check here? It seems that "foo" has never been `put` in the store so far?